core/dnsserver/register_test.go

```diff
@@ -29,3 +29,80 @@ func TestHandlers(t *testing.T) {
 		t.Errorf("Expected [testPlugin] from Handlers, got %v", hs)
 	}
 }
+
+func TestGroupingServers(t *testing.T) {
+	for i, test := range []struct {
+		configs []*Config
+		gpes    []string
+		failing bool
+	}{
+		{configs: []*Config{
```

Comment:
Do all these test cases really test new functionality? Seems to me you can get a away with 2 or 3 cases here?